src/platform/react-native.ts

```diff
@@ -0,0 +1,79 @@
+// React Native Platform Support
+import type { ICache } from '../types/Cache.js';
+import { Platform } from '../utils/Utils.js';
+import sha1Hash from './polyfills/web-crypto.js';
+import package_json from '../../package.json' assert { type: 'json' };
+import evaluate from './jsruntime/jinter.js';
+
+class Cache implements ICache {
+  #persistent_directory: string;
+  #persistent: boolean;
+
+  constructor(persistent = false, persistent_directory?: string) {
+    this.#persistent_directory = persistent_directory || '';
+    this.#persistent = persistent;
+  }
+
+  get cache_dir() {
+    return this.#persistent ? this.#persistent_directory : '';
+  }
+
+  #getStorage() {
+    const storage = new ((globalThis as any).mmkvStorage as any)({ id: 'InnertubeCache' });
```

Comment:
Might be worth throwing an error with an explanation when `mmkvStorage` is missing, that will prevent any cryptic errors further down the line.

Reply:
Agree

Reply:
@absidue makes sense! I've just left a https://github.com/LuanRT/YouTube.js/pull/593#issuecomment-2150285749 and if this is the only issue that is blocking as per @LuanRT consideration – I will get it fixed asap